app/scripts/modules/kubernetes/cluster/configure/CommandBuilder.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.kubernetes.clusterCommandBuilder.service', [
+  require('../../../core/config/settings.js'),
+  require('../../../core/account/account.service.js'),
+  require('../../../core/naming/naming.service.js'),
+  require('../../../core/utils/lodash.js'),
+])
+  .factory('kubernetesClusterCommandBuilder', function (settings, $q, accountService, namingService, _) {
+    function attemptToSetValidAccount(application, defaultAccount, command) {
+      return accountService.listAccounts('kubernetes').then(function(kubernetesAccounts) {
+        var kubernetesAccountNames = _.pluck(kubernetesAccounts, 'name');
+        var firstKubernetesAccount = null;
+
+        if (application.accounts.length) {
+          firstKubernetesAccount = _.find(application.accounts, function (applicationAccount) {
+            return kubernetesAccountNames.indexOf(applicationAccount) !== -1;
+          });
+        }
+
+        var defaultAccountIsValid = defaultAccount && kubernetesAccountNames.indexOf(defaultAccount) !== -1;
+
+        command.account =
+          defaultAccountIsValid ? defaultAccount : (firstKubernetesAccount ? firstKubernetesAccount : 'my-account-name');
+      });
+    }
+
+    function buildNewClusterCommand(application, defaults = {}) {
+      var defaultAccount = defaults.account || settings.providers.kubernetes.defaults.account;
+
+      var command = {
+        account: defaultAccount,
+        application: application.name,
+        strategy: '',
+        targetSize: 1,
+        cloudProvider: 'kubernetes',
+        selectedProvider: 'kubernetes',
+        namespace: 'default',
+        containers: [],
+        volumeSources: [],
+        buildImageId: buildImageId,
+        groupByRegistry: groupByRegistry,
+        viewState: {
+          mode: defaults.mode || 'create',
+          disableStrategySelection: true,
+        }
+      };
+
+      if (application && application.attributes && application.attributes.platformHealthOnly) {
+        command.interestingHealthProviderNames = ['Kubernetes'];
+      }
+
+      attemptToSetValidAccount(application, defaultAccount, command);
+
+      return command;
+    }
+
+    function buildClusterCommandFromExisting(application, existing, mode) {
+      mode = mode || 'clone';
+
+      var command = existing.deployDescription;
+
+      command.groupByRegistry = groupByRegistry;
+      command.cloudProvider = 'kubernetes';
+      command.selectedProvider = 'kubernetes';
+      command.account = existing.account;
+      command.buildImageId = buildImageId;
+      command.strategy = '';
+
+      command.containers.map((container) => {
```

Comment:
Since you're not returning anything here, `forEach` is more appropriate than `map`.